test/testcafe/spec/ChallengeAuthenticatorEmail_spec.js

```diff
@@ -576,37 +591,39 @@ test
 
 test
   .requestHooks(dynamicContinuePollingLogger, dynamicRefreshShortIntervalMock)('continue polling on form error with dynamic polling', async t => {
+    dynamicContinuePollingLogger.clear();
     const challengeEmailPageObject = await setup(t);
+    const waiter = createWaiter(t);
     await checkA11y(t);
     await challengeEmailPageObject.clickEnterCodeLink();
 
     await t.expect(await challengeEmailPageObject.resendEmailExists()).eql(false);
 
     // 2 poll requests in 2 seconds at 1 sec interval (Cumulative Request: 2)
-    await t.wait(2000);
+    await waiter.wait(1000);
+    // change mocks after 1st poll request so 2 sec interval will be applied after 2nd poll request
+    await t.removeRequestHooks(dynamicRefreshShortIntervalMock);
+    await t.addRequestHooks(invalidOTPMockContinuePoll);
+    await waiter.wait(1000);
     await t.expect(dynamicContinuePollingLogger.count(
       record => record.response.statusCode === 200 &&
         record.request.url.match(/poll/)
     )).eql(2);
-    dynamicContinuePollingLogger.clear();
```

Comment:
Reworked this flaky test.

Calling `logger.clear()` here makes no sense since it clears request count that is used in assertions below (`clear` was added in https://github.com/okta/okta-signin-widget/pull/3147)
Test relies on the fact that `await t.expect(logger.count(...)).eql(3)` will wait several seconds until request count reaches 3 ( from 0 after calling `clear()`)

Using `createWaiter` util here for more precise timers. 
Need to respect the fact that  ~1s can be elapsed between `await setup(t)` and first `wait()` call.
So it's a matter of luck to have poll request count of 2 (instead of 3 which can happen) at the time of `.eql(2)` check.
To make it stable we can rely on timestamp right after `await setup()`